yesod-form/Yesod/Form/Types.hs

```diff
@@ -45,6 +45,8 @@ import Data.Foldable
 --
 -- The 'Applicative' instance will concatenate the failure messages in two
 -- 'FormResult's.
+-- The 'Alternative' instance will choose 'FormFailure' before 'FormSuccess',
```

Comment:
Rereading this comment, I'm not sure it's true (though maybe my "cleanup" broke the intended behavior :)). There doesn't seem to be any preference for `FormFailure` over `FormSuccess` in the instance, only `FormMissing` has special handling.